index.js

```diff
@@ -142,12 +142,39 @@ class HdKeyring extends EventEmitter {
     return Promise.resolve(sig)
   }
 
-  // personal_signTypedData, signs data along with the schema
-  signTypedData (withAccount, typedData) {
-    const wallet = this._getWalletForAccount(withAccount)
-    const privKey = ethUtil.toBuffer(wallet.getPrivateKey())
-    const signature = sigUtil.signTypedData(privKey, { data: typedData })
-    return Promise.resolve(signature)
+  signTypedData (address, typedData, opts = { version: 'V1' }) {
```

Comment:
i assume this is called externally, so it'd be good to have a unit test for it.

Reply:
sounds good, will add one 👍